business/checkers/gateways/multi_match_checker.go

```diff
@@ -161,3 +163,17 @@ func (m MultiMatchChecker) findMatch(host Host, selector string) (bool, []Host)
 	}
 	return len(duplicates) > 0, duplicates
 }
+
+func regexpFromHostname(hostname string) *regexp.Regexp {
+	// Escaping dot chars for RegExp. Dot char means all possible chars.
+	// This protects this validation to false positive for (api-dev.example.com and api.dev.example.com)
+	escaped := strings.Replace(hostname, ".", "\\.", -1)
+
+	// We anchor the beginning and end of the string when it's
+	// to be used as a regex, so that we don't get spurious
+	// substring matches, e.g., "example.com" matching
+	// "foo.example.com".
+	anchored := strings.Join([]string{"^", escaped, "$"}, "")
```

Comment:
I thought that using `fmt.Sprintf("^%s$", escaped)` would go faster but it doesn't. Actually, it make some sense because the fmt needs to find characters within a string while the Join just appends whatever in the array into the string. No lookup necessary. I guess.
So no changes need it.